src/test/java/io/kokuwa/maven/helm/junit/MojoExtension.java

```diff
@@ -1,166 +1,113 @@
-		FileUtils.forceDelete(getProjectBuildDirectory(context).toFile());
-	}
-
 	// parameter
 
 	@Override
-	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext context)
-			throws ParameterResolutionException {
-		return AbstractHelmMojo.class.isAssignableFrom(parameterContext.getParameter().getType());
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext context) {
+		return mojos.containsKey(parameterContext.getParameter().getType());
 	}
 
 	@Override
-	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext context)
-			throws ParameterResolutionException {
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext context) {
 		try {
 
 			// get descriptor
 
-			Class<? extends AbstractHelmMojo> mojoType = (Class<AbstractHelmMojo>) parameterContext.getParameter()
-					.getType();
-			MojoDescriptor descriptor = mojoDescriptors.get(mojoType);
-			assertNotNull(descriptor, "Plugin descriptor for " + mojoType.getSimpleName()
-					+ " not found, run 'maven-plugin-plugin:descriptor'.");
+			Class<AbstractHelmMojo> mojoType = (Class<AbstractHelmMojo>) parameterContext.getParameter().getType();
+			MojoDescriptor descriptor = mojos.get(mojoType);
+			assertNotNull(descriptor, "Plugin descriptor for " + mojoType.getSimpleName() + " not found.");
 
-			// create mojo with default values
+			// create mojo
 
 			AbstractHelmMojo mojo = spy(mojoType);
-			for (Parameter parameter : descriptor.getParameters()) {
-				if (parameter.getDefaultValue() == null || !parameter.isEditable()
-						|| parameter.getType().equals("boolean")) {
-					continue;
-				}
-				getField(mojoType, parameter.getName()).set(mojo, resolve(context, parameter.getDefaultValue()));
-			}
-
-			// read mojo values from annotations
-
-			MojoProperty[] mojoProperties = ArrayUtils.addAll(
-					context.getRequiredTestClass().getAnnotationsByType(MojoProperty.class),
-					context.getRequiredTestMethod().getAnnotationsByType(MojoProperty.class));
-			for (MojoProperty mojoProperty : mojoProperties) {
-				Field field = getField(mojoType, mojoProperty.name());
-				field.set(mojo, field.getType().equals(URL.class)
-						? new URL(mojoProperty.value())
-						: resolve(context, mojoProperty.value()));
-			}
-
-			// settings
+			mojo.setSettings(new Settings());
+			mojo.setSecurityDispatcher(new DefaultSecDispatcher(new DefaultPlexusCipher()));
+			mojo.setLog(new SystemStreamLog());
+			mojo.setChartDirectory("src/test/resources/simple"); // set some sane defaults for tests
+			mojo.setHelmVersion("3.10.0"); // avoid github api
 
-			getField(mojoType, "settings").set(mojo, new Settings());
+			// set parameter
 
-			// plexus SecDispatcher
-
-			SecDispatcher secDispatcher = spy(new DefaultSecDispatcher(new DefaultPlexusCipher()));
-			getField(mojoType, "securityDispatcher").set(mojo, secDispatcher);
+			for (Parameter parameter : descriptor.getParameters()) {
```

Comment:
How did you get this nesting through your checkstyle;)

Reply:
i do not get your point